src/registrar/management/commands/utility/extra_transition_domain_helper.py

```diff
@@ -0,0 +1,1022 @@
+                options.domain_escrow_filename,
+                DomainEscrow,
+                "domainname",
+            ),
+            (
+                EnumFilenames.DOMAIN_ADHOC,
+                options.domain_adhoc_filename,
+                DomainTypeAdhoc,
+                "domaintypeid",
+            ),
+            (
+                EnumFilenames.ORGANIZATION_ADHOC,
+                options.organization_adhoc_filename,
+                OrganizationAdhoc,
+                "orgid",
+            ),
+            (
+                EnumFilenames.AUTHORITY_ADHOC,
+                options.authority_adhoc_filename,
+                AuthorityAdhoc,
+                "authorityid",
+            ),
+            (
+                EnumFilenames.AUTHORITY_ADHOC,
+                options.authority_adhoc_filename,
+                AuthorityAdhoc,
+                "authorityid",
+            ),
+        ]
+
+        self.file_data = self.populate_file_data(pattern_map_params)
+
+    # TODO - revise comment
+    def populate_file_data(self, pattern_map_params):
+        """Populates the self.file_data field given a set
+        of tuple params.
+
+        pattern_map_params must adhere to this format:
+            [
+                (file_type, filename, data_type, id_field),
+            ]
+
+        vars:
+            file_type (EnumFilenames) -> The name of the dictionary.
+            Defined as a value on EnumFilenames, such as
+            EnumFilenames.AGENCY_ADHOC
+
+            filename (str) -> The filepath of the given
+            "file_type", such as migrationdata/test123.txt
+
+            data_type (type) -> The type of data to be read
+            at the location of the filename. For instance,
+            each row of test123.txt may return data of type AgencyAdhoc
+
+            id_field (str) -> Given the "data_type" of each row,
+            this specifies what the "id" of that row is.
+            For example, "agencyid". This is used so we can
+            store each record in a dictionary rather than
+            a list of values.
+
+        return example:
+            EnumFilenames.AUTHORITY_ADHOC: FileDataHolder(
+                authority_adhoc_filename,
+                self.strip_date_regex,
+                AuthorityAdhoc,
+                "authorityid",
+            ),
+        """
+        file_data = {}
+        for file_type, filename, data_type, id_field in pattern_map_params:
+            file_data[file_type] = FileDataHolder(
+                filename,
+                self.strip_date_regex,
+                data_type,
+                id_field,
+            )
+        return file_data
+
+    def parse_all_files(self, infer_filenames=True):
+        """Clears all preexisting data then parses each related CSV file.
+
+        infer_filenames: bool -> Determines if we should try to
+        infer the filename if a default is passed in
+        """
+        self.clear_file_data()
```

Comment:
**question (non-blocking)**: This is only ever called from `__init__`. Why do we need to clear data here?

Reply:
Good call. The original intent was to allow the script to attempt to conditionally re-parse under certain conditions (like renaming a file on the fly) without having to reload the entire script but that didn't get implemented. Code remnant